src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -76888,5 +76900,1310 @@ A M()
             var comp = CreateCompilation(source);
             comp.VerifyDiagnostics();
         }
+
+        [Fact]
+        public void Deconstruction_01()
+        {
+            var source =
+@"class Program
+{
+    static void F<T, U>() where U : class
+    {
+        (T x, U y) = default((T, U)); // 1
+        x.ToString(); // 2
+        y.ToString(); // 3
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (5,22): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         (T x, U y) = default((T, U)); // 1
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "default((T, U))").WithLocation(5, 22),
+                // (6,9): warning CS8602: Possible dereference of a null reference.
+                //         x.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(6, 9),
+                // (7,9): warning CS8602: Possible dereference of a null reference.
+                //         y.ToString(); // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "y").WithLocation(7, 9));
+        }
+
+        [Fact]
+        public void Deconstruction_02()
+        {
+            var source =
+@"class Program
+{
+    static void F<T, U>() where U : class
+    {
+        (T x, U y) = (default, default); // 1, 2
+        x.ToString(); // 3
+        y.ToString(); // 4
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (5,23): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         (T x, U y) = (default, default); // 1, 2
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "default").WithLocation(5, 23),
+                // (5,32): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         (T x, U y) = (default, default); // 1, 2
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "default").WithLocation(5, 32),
```

Comment:
Why are there two diagnostics here and only one above? Logically this seems like the same operation.

Reply:
The deconstruction is treated as two separate assignments: `x = default; y = default;`.